### DISCUSSION TITLE
Getting the Player's current playback position

### DISCUSSION THREAD

**Discussion by RainbowCookie32** [librespot-org/librespot#921] [General]
Hi! I've been using librespot for a while now, and I've always wondered: is there a way to reliably get the player's playback position?

The `Playing` PlayerEvent message reports the current position, but as the docs for it mention, it's only sent in some pretty specific situations. I looked around a bit but I can't see anything that'd return this value. Is there any way to get it that I'm not seeing?

**Comment by roderickvd** [comment]
Only within the player itself in the `stream_position_pcm` fields. If you need to do anything with this value then I'd recommend implementing your own player rather than sending event messages on each packet.
